cdap-ui/app/cdap/components/DataPrepConnections/DatabaseBrowserWrapper/index.js

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { Component, PropTypes } from 'react';
+import DataPrepBrowser from 'components/DataPrep/DataPrepBrowser';
+import {setActiveBrowser, setDatabaseProperties} from 'components/DataPrep/DataPrepBrowser/DataPrepBrowserStore/ActionCreator';
+import NamespaceStore from 'services/NamespaceStore';
+import MyDataPrepApi from 'api/dataprep';
+import {objectQuery} from 'services/helpers';
+import LoadingSVGCentered from 'components/LoadingSVGCentered';
+
+export default class DatabaseBrowserWrapper extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      id: null,
+      loading: true,
+      error: null
+    };
+
+    this.fetching = false;
+  }
+
+  componentWillMount() {
+    this.fetchConnectionInfo(this.props.databaseId);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.databaseId === this.state.id) { return; }
+    this.fetchConnectionInfo(nextProps.databaseId);
+  }
+
+  fetchConnectionInfo(id) {
+    if (this.fetching) { return; }
```

Comment:
Wouldn't this be a problem if user clicks on a different connection than he is on but the response for it is not returned by the backend yet. Since we check for databaseId to be different shouldn't this always render the latest database details upon click on the latest databaseId?